lib/Sema/TypeCheckAccess.cpp

```diff
@@ -2552,7 +2552,7 @@ void swift::diagnoseUnnecessaryPublicImports(SourceFile &SF) {
                                           diag::remove_package_import;
     auto inFlight = ctx.Diags.diagnose(import.importLoc,
                                        diagId,
-                                       importedModule);
+                                       importedModule->getName());
```

Comment:
Do we not need to use the public name here?

Reply:
We want the real module name here, the one that matches the import statement. This would be the diagnostics on a module with an `-public-module-name` otherwise:

```
public import SwiftPublicNameCore // expected-warning {{public import of 'SwiftPublicName' was not used in public declarations or inlinable code}}
public import SwiftPublicName
```

This is not covered anymore since I removed a commit. If I have to fix more things on this PR I'll look to add this back in the test.